### ISSUE TITLE
Possible concurrent test issue with nightly live tests

### ISSUE THREAD

**Issue by heaths** [Azure/azure-sdk-for-net#10496]
Nightlies failed with the error:

```
Status: 404 (Not Found)

Content:

{"error":{"code":"SecretNotFound","message":"Secret not found: 743893227"}}

```

The stack was:

```
   at Azure.Security.KeyVault.KeyVaultPipeline.SendRequestAsync(Request request, CancellationToken cancellationToken) in /home/vsts/work/1/s/sdk/keyvault/Azure.Security.KeyVault.Shared/src/KeyVaultPipeline.cs:line 240
   at Azure.Security.KeyVault.KeyVaultPipeline.SendRequestAsync[TResult](RequestMethod method, Func`1 resultFactory, CancellationToken cancellationToken, String[] path) in /home/vsts/work/1/s/sdk/keyvault/Azure.Security.KeyVault.Shared/src/KeyVaultPipeline.cs:line 191
   at Azure.Security.KeyVault.Secrets.SecretClient.GetSecretAsync(String name, String version, CancellationToken cancellationToken) in /home/vsts/work/1/s/sdk/keyvault/Azure.Security.KeyVault.Secrets/src/SecretClient.cs:line 93
   at Azure.Security.KeyVault.Secrets.Samples.GetSecrets.GetSecretsAsync() in /home/vsts/work/1/s/sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/samples/Sample3_GetSecretsAsync.cs:line 53
   at Azure.Security.KeyVault.Secrets.Samples.GetSecrets.GetSecretsAsync() in /home/vsts/work/1/s/sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/samples/Sample3_GetSecretsAsync.cs:line 45
   at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter`1.BlockUntilCompleted()
   at NUnit.Framework.Internal.AsyncToSyncAdapter.Await(Func`1 invoke)
   at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
   at NUnit.Framework.Internal.Execution.SimpleWorkItem.PerformWork()
```

Since these are samples running, they may be running concurrently with other live tests since they don't share a common fixture.

**Comment by heaths** [comment]
While the samples and live tests are both attributed with `[NonParallelizable]`, according to [NUnit](https://github.com/nunit/docs/wiki/Engine-Parallel-Test-Execution) assemblies are run in parallel, which could explain the race condition. I'm digging into our test infrastructure to see if/how I can pass `--agents` and NUnit documentation to see if/how I can use `ProcessModel.Single` it refers to (but I can find nowhere in code or other docs).

**Comment by heaths** [comment]
This was solved with PR #11170.
